src/recipe/parser/test.rs

```diff
@@ -304,18 +304,38 @@ impl TryConvertNode<PythonVersion> for RenderedNode {
                 *self.span(),
                 ErrorKind::InvalidField("expected string, sequence or null".into()),
             )])?,
-            RenderedNode::Scalar(version) => PythonVersion::Single(version.to_string()),
+            RenderedNode::Scalar(version) => {
+                let version_str = version.to_string();
+                if rattler_conda_types::MatchSpec::from_str(&version_str, rattler_conda_types::ParseStrictness::Strict).is_err() {
+                    return Err(vec![_partialerror!(
+                        *self.span(),
+                        ErrorKind::InvalidValue(("python_version".to_string(), "Invalid version specification".into())),
+                        label = format!("invalid python version specification: {}", version_str),
+                        help = "Use a valid version specification like '3.12', '3.12.*', or '>=3.12'"
```

Comment:
I am quite sure that `3.12` will not parse with the Strict mode, so we should not mention it here :) You could mention `==3.12` but I don't think that's what most people want.